torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1530,12 +1593,21 @@ def _verify(expr, potential_expr):
             for i, ss in enumerate(t.size()):
                 property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
                 track_symint(property_source, ss)
-                if _is_dim_dynamic(t, i):
-                    # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
-                    # constrained to an integer.
-                    if _is_int(ss):
-                        raise RuntimeError(f"Attempting to constrain dim {i} for {source}, which violates user's mark_dynamic")
-                    dynamic_sources.append(property_source)
+                if dynamic_ranges:
+                    dyn_range = dynamic_ranges[pos]
+                    if dyn_range and i in dyn_range:
+                        # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
+                        # constrained to an integer.
+                        if _is_int(ss):
+                            raise RuntimeError(f"Attempting to constrain dimension "
+                                               f"{source.name()}.size()[{i}] to {int(ss)}, "
+                                               "which violates user's constraints")
+
+                        vr = dyn_range[i].to_range()
+                        if vr != _default_value_range(specialize_zero_one=self.specialize_zero_one):
+                            for symbol in ss.node.expr.free_symbols:
+                                self._verify_valid_range(symbol, vr)
+                        dynamic_sources.append(property_source)
```

Comment:
This still seems weirdly complicated

Reply:
What do you prefer?

Reply:
I'm not sure. I haven't understood what the code is doing yet